Dockerfile

```diff
@@ -1,3 +1,7 @@
 FROM nginx:1.15-alpine
 COPY dist/hypertrace-ui /usr/share/nginx/html
-COPY conf/default.conf /etc/nginx/conf.d/default.conf
\ No newline at end of file
+COPY conf/default.conf /etc/nginx/conf.d/default.conf
+# expose port for UI
+EXPOSE 2020
+# Healthcheck to check if client can connect to UI port
+HEALTHCHECK --interval=1s --retries=5 --timeout=1s CMD wget -qO- http://127.0.0.1:2020 &> /dev/null || exit 1
```

Comment:
I think this will be costly, will this `http://127.0.0.1:2020` try to execute default query to fetch dashboard. I think, here it would be fine just to check port is registered.
`nc -z 127.0.0.1 2020`

Reply:
before the command was to check the proxy pass operates. I think we can find a cheap test for that which doesn't produce a lot of load https://github.com/hypertrace/hypertrace-ui/blob/master/conf/default.conf#L10-L12

nc isnt useful in nginx as it will almost always pass especially as port config is static.

Reply:
it was `wget http://localhost:2020/graphql\?query\=\{metadata\{name\}\}` at some point..